src/setup.rs

```diff
@@ -18,19 +18,31 @@ const VERSION: &str = env!("CARGO_PKG_VERSION");
 /// Set by our `build.rs`, reflects the Rust target triple we're building for
 const TARGET: &str = env!("TARGET");
 
-/// Where Kani has been installed. Typically `~/.kani/kani-1.x/`
-pub fn kani_dir() -> PathBuf {
-    home::home_dir().expect("Couldn't find home dir?").join(".kani").join(format!("kani-{VERSION}"))
+/// The directory where Kani is installed, either:
+///  * `${KANI_HOME}/.kani/kani-<VERSION>` if the environment variable
+///    `KANI_HOME` is set.
+///  * `${HOME}/.kani/kani-<VERSION>` where `HOME` is the canonical definition
+///    of home directory used by Cargo and rustup.
+pub fn kani_dir() -> Result<PathBuf> {
+    let home_dir = match env::var("KANI_HOME") {
+        Ok(val) => PathBuf::from(val),
+        Err(_) => home::home_dir().expect("couldn't find home directory"),
+    };
+    if !home_dir.is_dir() {
+        bail!("got home directory which isn't a directory");
```

Comment:
Can you please improve the error message to include the value of `home_dir`? You can also include whether you got it from KANI_HOME.